CSS/HTML and CSS Basics/css-basics/css-specificity.md

```diff
@@ -28,96 +28,73 @@ links:
-id priority > class priority > type priority
-```
-The last step in calculating a selector's specificity is to see the above as the header of a table with three columns:
-```
-         id - class - type
-selector  0 -   0   -  0  
-```
-For each basic selector that is part of a more complex one, we increment the value in the appropriate column.
-
-For example, consider the following selector that aims to style a heading, nested inside a div of class *my-class*, which is another div's child:
-```css
-div div.my-class h2{}
+Sometimes *CSS* does not behave in the way we want or expect it to, suppose we have this piece of code:
 ```
-There are three *type* selectors (*div, div and h2*) and one *class* selector (*.my-class*). We can use the previous syntax to represent the **specificity** of the selector:
-```text
-id - class - type
-0  -   1   -  3
+HTML:
+<ul id="dream-garage">
+   <li>Saleen S7R</li>
+   <li>Lamborghini Countach</li>         
+   <li>Ferrari F430 GT3</li>
+</ul>
 ```
-However, this should not be seen as a base 10 representation. Adding an id selector on top of 99 class selectors would yield:
-```text
-         id - class - type
-selector  1 -   99   -  0
+And now we want to hightlight the favourite one:
 ```
-Moreover, when comparing two selectors, you should always be aware of one thing: it doesn't matter how many type or class basic selectors make up a more complex one, they will always be outweighed by a class or an id selector, respectively:
-```text
-         id - class - type
-selector1  0 - 110  -  0
-selector2  1 -   0  -  0
+HTML:
+<ul id="dream-garage">
+   <li class="favourite">Saleen S7R</li>
+   <li>Lamborghini Countach</li>
+   <li>Ferrari F430 GT3</li>
+</ul>
+CSS:
+.favorite {
+  color: red;
+  font-weight: bold;
+}
 ```
-In the above example, *selector2*'s styling is applied in case they target the same elements.
-
-Let's go back to the first example. Say we decide to give the heading the id of *title*:
-```css
-div div.my-class #title{}
+But upon compilation we don't get our Saleen highlighted. And after looking through *CSS* document we find:
 ```
-And use two selectors to target the title of the page:
-```css
-div div.my-class #title{
-  font-size: 30px;
-}
-div div.my-class h2{
-  font-size: 40px;
+ul#dream-garage li {
+   font-weight: normal;                   
+   font-size: 12px;
+   color: black;
 }
 ```
-If cascading alone was the way to go, the heading *font-size* would be 40. Taking specificity into account:
-```text
-                        id - class - type
-div div.my-class #title  1 -   1   -  2
-div div.my-class h2      0 -   1   -  3
-```
-The font size is *30*.
-
-As discussed earlier, inline CSS has the highest inherited specificity:
-```html
-<div>
-  <div class="my-class">
-    <h2 id="title"
-      style="font-size:20px;">
-      Title
-    </h2>
-  </div>
-</div>
+In this case *id "dream-garage"* selector is more **specific** than *class "favourite"* which neutralises effect of the latter.
```

Comment:
In this case, the id selector "dream-garage" is more specific than the "favourite" class, which effectively overrides the effect of the later.